c++/src/capnp/compat/http-over-capnp.c++

```diff
@@ -323,6 +323,37 @@ private:
   // Must check state->assertNotCanceled() before using this.
 };
 
+class HttpOverCapnpFactory::ConnectClientRequestContextImpl final
+    : public capnp::HttpService::ConnectClientRequestContext::Server {
+public:
+  ConnectClientRequestContextImpl(HttpOverCapnpFactory& factory,
+                           kj::HttpService::ConnectResponse& connResponse)
+      : factory(factory), connResponse(connResponse) {}
+
+  kj::Promise<void> startConnect(StartConnectContext context) override {
+    KJ_REQUIRE(!sent, "already called startConnect()");
+    sent = true;
+
+    auto params = context.getParams();
+    auto resp = params.getResponse();
+
+    auto headers = factory.headersToKj(resp.getHeaders());
+    if (resp.getIsAccept()) {
+      connResponse.accept(resp.getStatusCode(), resp.getStatusText(), headers);
+    } else {
+      connResponse.reject(resp.getStatusCode(), resp.getStatusText(), headers);
```

Comment:
This returns a stream, to which you're supposed to write the error body, but instead you're discarding it.

Looking at the server side, it appears that you are writing the error body into the socket itself. I guess that was based on an earlier version of James' PR, that took that approach. However, we decided this was confusing -- the application is not expecting an error body (e.g., some HTML error page) to appear on the connection stream where it was expecting to speak some other protocol (e.g., PostgreSQL protocol). We need to keep these streams separtae.

`ConnectClientRequestContext` should probably have a second method for this purpose, like:

```
startError @0 (response :HttpResponse) -> (body :ByteStream);
```

Alternatively, perhaps `ConnectClientRequestContext` should actually be merged with `ClientRequestContext`. For the error case, you would use the regular `startRespnose()`. `startConnect()` would then be similar to `startWebSocket()` -- a method that is only called if the request was a valid connect request.

Reply:
I will keep the request contexts separate, mainly because there are two different HttpService::Response types for each (`connect` uses `HttpService::ConnectResponse`).